core/server/master/src/main/java/alluxio/master/meta/MetaDailyBackup.java

```diff
@@ -0,0 +1,182 @@
+import alluxio.wire.BackupOptions;
+import alluxio.wire.BackupResponse;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.TreeMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Backing up primary master metadata everyday at a fixed UTC time.
+ */
+public final class MetaDailyBackup {
+  private static final Logger LOG = LoggerFactory.getLogger(MetaDailyBackup.class);
+
+  private final Pattern mBackupPattern
+      = Pattern.compile("alluxio-backup-[0-9]+-[0-9]+-[0-9]+-([0-9]+).gz");
+  private final int mMaxFile = Configuration.getInt(PropertyKey.MASTER_DAILY_BACKUP_FILE_MAX);
+
+  private ScheduledFuture<?> mBackup;
+  private MetaMaster mMetaMaster;
+  private ScheduledExecutorService mScheduleExecutor;
+
+  /**
+   * Constructs a new {@link MetaDailyBackup}.
+   *
+   * @param metaMaster the meta master
+   */
+  MetaDailyBackup(MetaMaster metaMaster) {
+    mMetaMaster = metaMaster;
+  }
+
+  /**
+   * Starts {@link MetaDailyBackup}.
+   */
+  public void start() {
+    Preconditions.checkState(mBackup == null && mScheduleExecutor == null);
+    mScheduleExecutor = Executors.newSingleThreadScheduledExecutor(
+    ThreadFactoryUtils.build("MetaDailyBackup-%d", true));
+
+    mBackup = mScheduleExecutor.scheduleAtFixedRate(new Runnable() {
+          @Override
+          public void run() {
+            dailyBackup();
+          }
+        }, getTimeToNextBackup(), FormatUtils.parseTimeSize("1day"),
+        TimeUnit.MILLISECONDS);
+    LOG.info("MetaDailyBackup scheduled.");
+  }
+
+  /**
+   * Gets the time gap between now and next backup time.
+   *
+   * @return the time gap to next backup
+   */
+  private long getTimeToNextBackup() {
+    LocalDateTime now = LocalDateTime.now(Clock.systemUTC());
+    String[] hourAndMin = Configuration.get(PropertyKey.MASTER_DAILY_BACKUP_TIME).split(":");
+    int hour = Integer.parseInt(hourAndMin[0]);
+    int min = hourAndMin.length == 2 ? Integer.parseInt(hourAndMin[1]) : 0;
+    LocalDateTime nextBackupTime = LocalDateTime.of(now.getYear(),
+        now.getMonth(), now.getDayOfMonth(), hour, min, 0);
+    if (nextBackupTime.isBefore(now)) {
+      nextBackupTime = nextBackupTime.plusDays(1);
+    }
+    return ChronoUnit.MILLIS.between(now, nextBackupTime);
+  }
+
+  /**
+   * The daily backup task.
+   */
+  private void dailyBackup() {
+    String dir = Configuration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
+    String rootUfs = Configuration.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
+    boolean isLocal = URIUtils.isLocalFilesystem(Configuration
```

Comment:
do we need to give a local filesystem option, or should we only use root ufs?

Reply:
I added support for local filesystem absolute paths. Thanks